hazelcast/src/hazelcast/client/sql.cpp

```diff
@@ -828,6 +840,100 @@ sql_result::page_iterator::has_next() const
     return !*last_;
 }
 
+sql_result::page_iterator_sync::non_copyables::non_copyables(
+  page_iterator&& iter)
+  : preloaded_page_{}
+  , iter_{ std::move(iter) }
+{
+}
+
+sql_result::page_iterator_sync::page_iterator_sync(
```

Comment:
I suggest this method to be private and sql_result to be a friend of this class. It seems user does not need it.

Reply:
Fixed.